src/Juvix/Compiler/Core/Data/BinderList.hs

```diff
@@ -8,12 +10,18 @@ data BinderList a = BinderList
   { _blLength :: Int,
     _blMap :: [a]
   }
+  deriving stock (Eq, Generic)
+
+instance NFData a => NFData (BinderList a)
 
 makeLenses ''BinderList
 
 fromList :: [a] -> BinderList a
 fromList l = BinderList (length l) l
 
+empty :: BinderList a
+empty = fromList []
```

Comment:
There is `mempty` already. This should not be duplicated with `empty`.

Reply:
Done.